src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java

```diff
@@ -147,6 +151,9 @@ public String getType() {
     @JoinColumn(name = "iris_settings_id", table = "programming_exercise_details")
     private IrisSettings irisSettings;
 
+    @Column(name = "build_plan_configuration", table = "programming_exercise_details", columnDefinition = "json")
+    private String buildPlanConfiguration;
```

Comment:
For other build plans we’ve already got a specific table. Would it make sense to instead add a discriminator (example how that works should be `Exercise` and subclasses) and the `build_plan_configuration` as columns and then store the AeolusBuildPlans as different specific type of build plans?
That might make it easier to extend the build plan definition in the future with additional metadata if necessary.

The class structure would then be something like
```java
// split up the current build plan into the common part and the build plan text definition
@DiscriminatorColumn(
    name="build_plan_type",
    discriminatorType = DiscriminatorType.CHAR // or STRING?
)
abstract class BuildPlan extends DomainObject {
     ProgrammingExercise exercise;
}

@DiscriminatorValue("N")
class NativeBuildPlan extends BuildPlan {
     String buildPlan;
}

// add the Aeolus build plan that stores the build plan differently
@DiscriminatorValue("A")
class AeolusBuildPlan extends BuildPlan {
     @Column(name = "build_plan_configuration", table = "programming_exercise_details", columnDefinition = "json")
     String buildPlanConfiguration;
}
```

Reply:
Thanks for the idea, but I would rather keep it as-is.
I think specifying a column of a table in a different class/file makes it harder to understand what properties a class really has. I also think updating with additional metadata seems the same to me in both approaches. So if it's non-blocking for you I would love to keep it as an additional value of the programming exercise.

Reply:
Fine by me.

Keeping the conversation open in case a maintainer has an opinion as part of the database migration review.

Reply:
I just spoke with @krusche, and based on my explanation, he said it's fine the way I implemented it.